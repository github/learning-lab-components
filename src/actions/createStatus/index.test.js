const createStatus = require('./')
const mockContext = require('../../tests/mockContext')

describe('createStatus', () => {
  let context

  beforeEach(() => {
    context = mockContext({
      pull_request: {
        body: 'This here is a body',
        head: { sha: 123 }
      }
    }, {
      repos: {
        createStatus: jest.fn()
      }
    })
  })

  it('creates a status', async () => {
    await createStatus(context, { state: 'pending' })
    expect(context.github.repos.createStatus).toHaveBeenCalled()
    expect(context.github.repos.createStatus.mock.calls).toMatchSnapshot()
  })

  it('creates a status with expected state', async () => {
    await createStatus(context, { state: 'success' })
    expect(context.github.repos.createStatus).toHaveBeenCalled()
    expect(context.github.repos.createStatus.mock.calls).toMatchSnapshot()
  })

  it('creates a status with a gate-ish string', async () => {
    await createStatus(context, { state: '%payload.pull_request.body%' })
    expect(context.github.repos.createStatus).toHaveBeenCalled()
    expect(context.github.repos.createStatus.mock.calls).toMatchSnapshot()
  })

  it('creates a status with a sub-object', async () => {
    context.payload.pull_request.body = ''
    await createStatus(context, { state: '%payload.pull_request.body%', failure: { description: 'Your status' } })
    expect(context.github.repos.createStatus).toHaveBeenCalled()
    expect(context.github.repos.createStatus.mock.calls).toMatchSnapshot()
  })
})
