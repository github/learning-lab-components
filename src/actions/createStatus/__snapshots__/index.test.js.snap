// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createStatus creates a status 1`] = `
Array [
  Array [
    Object {
      "context": "my-step",
      "owner": "JasonEtco",
      "repo": "example",
      "sha": 123,
      "state": "pending",
    },
  ],
]
`;

exports[`createStatus creates a status with a gate-ish string 1`] = `
Array [
  Array [
    Object {
      "context": "my-step",
      "owner": "JasonEtco",
      "repo": "example",
      "sha": 123,
      "state": "success",
    },
  ],
]
`;

exports[`createStatus creates a status with a sub-object 1`] = `
Array [
  Array [
    Object {
      "context": "my-step",
      "description": "Your status",
      "owner": "JasonEtco",
      "repo": "example",
      "sha": 123,
      "state": "failure",
    },
  ],
]
`;

exports[`createStatus creates a status with expected state 1`] = `
Array [
  Array [
    Object {
      "context": "my-step",
      "owner": "JasonEtco",
      "repo": "example",
      "sha": 123,
      "state": "success",
    },
  ],
]
`;
